features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentOptions.jsx

```diff
@@ -145,8 +145,17 @@ class CommentOptions extends React.Component {
         const {
             classes, comment, theme,
         } = this.props;
-        const canDelete = (comment.createdBy === AuthManager.getUser().name)
-            || (AuthManager.getUser().name === theme.custom.adminRole);
+        const user = AuthManager.getUser();
+        let username = null;
+        if (user) {
+            username = user.name;
+            const count = (username.match(/@/g) || []).length;
+            if (user.name.endsWith('@carbon.super') && count <= 1) {
```

Comment:
This same logic is used in  [here](https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Base/Header/avatar/Avatar.jsx#L88-L90) as well. Shall we use this to the [utils class](https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/Utils.js)

Reply:
Fixed